### DISCUSSION TITLE
using two cores to transfer files and improve file upload speed

### DISCUSSION THREAD

**Discussion by liqijian101** [luc-github/ESP3D#930] [Ideas]
One core is used to receive data, and the other is used to store data。
Here is a successful example。

https://www.bilibili.com/video/BV1vy4y1o7DB/?spm_id_from=333.999.0.0

**Comment by luc-github** [comment]
I am not sure to understand : 
>One core is used to receive data, and the other is used to store data。

On ESP32 with Marlin installed and ESP3DLib - Core 0  is used for wifi and related feature when Core 1 us used for Marlin itself
On ESP3D + ESP32 : when streaming will be enabled WiFi and related feature will be used by core 0 and streaming by Core 1

Is that what you mean ? Sorry watching video did not clarify your meaning - my chinese is limited ^_^ and I do not use cura so not sure about impact

**Comment by liqijian101** [comment]
yes

**Comment by luc-github** [comment]
but that is what is already implemented in ESP3DLib and planned for streaming support which is still under dev currently
